src/main/java/de/doubleslash/keeptime/view/SettingsController.java

```diff
@@ -538,4 +617,66 @@ private void showLicense(final Licenses license) {
          alert.show();
       }
    }
+
+   private void handleApiOff() {
+      Map<String, String> propertiesToUpdate = new HashMap<>();
+      setWebApplicationType("none");
+      propertiesToUpdate.put("api", "OFF");
+      propertyWrite(propertiesToUpdate);
+   }
+
+   private void handleApiOn() {
+      String username = authName.getText();
+      String password = authPassword.getText();
+
+      LoginController loginController = new LoginController(username, password);
+
+      loginController.createAndSaveUser();
+
+      Map<String, String> propertiesToUpdate = new HashMap<>();
+      propertiesToUpdate.put("spring.main.web-application-type", "");
+      propertiesToUpdate.put("server.port", authPort.getText());
+      propertiesToUpdate.put("api", "ON");
+
+      propertiesToUpdate.put("spring.security.user.name", "${BASIC_AUTH_USER:" + username + "}");
+      propertiesToUpdate.put("spring.security.user.password", "${BASIC_AUTH_PASSWORD:" + password + "}");
+      propertyWrite(propertiesToUpdate);
+   }
+
+   private void setWebApplicationType(String value) {
+      propertyWrite("spring.main.web-application-type", value);
+   }
+
+   public void propertyWrite(String key, String value) {
+      Properties properties = new Properties();
+      try (InputStream inputStream = getClass().getClassLoader().getResourceAsStream(propertiesFilePath);
+            FileOutputStream outputStream = new FileOutputStream(propertiesFilePath)) {
+
+         properties.load(inputStream);
+         properties.setProperty(key, value);
+         properties.store(outputStream, null);

```

Comment:
the .properties file does contain all properties again. we wanted to only have user overwritten rest-api properties in the additional properties file and not all properties like h2 version, I think also that this was working already before.

Reply:
it still puts all properties in the file